src/structure.js

```diff
@@ -81,6 +82,20 @@ function Structure (options) {
 inherits(Structure, EventEmitter);
 module.exports = Structure;
 
+Structure.prototype._subscribe = function(path, fn) {
```

Comment:
Could these (`_subscribe` and `_unsubscribe`) possibly be just stand alone functions, instead of exposing them?

E.g.

```js
function subscribe(listeners, path, fn) {
  return pathCombinations(path).reduce(function(listeners, path) {
    return listeners.updateIn([Immutable.List(path)], Immutable.Set(), function(old) {
      return old.add(fn);
    });
  }, listeners)
};
```